### PR TITLE
`Packs` pack cleanup

### PR DESCRIPTION
**Pull request by manasdk**
* Shallow clone the repo
* Rename class and refactor to make linter happy
* Add config headers

### CODE REVIEW

File: contrib/packs/actions/pack_mgmt/download.py [hunk 1]
```diff
@@ -141,7 +162,7 @@ def _eval_subtree(repo_url, subtree):
     @staticmethod
     def _eval_repo_url(repo_url):
         """Allow passing short GitHub style URLs"""
-        if len(repo_url.split('/')) == 2 and not "git@" in repo_url:
+        if len(repo_url.split('/')) == 2 and "git@" not in repo_url:
```
Comment (lakshmi-kannan) [inline contrib/packs/actions/pack_mgmt/download.py]: lol @ syntax blooper.

**Comment by lakshmi-kannan** [comment]
I forgot. I made this change a while ago and didn't push :P. LGTM.

**Comment by Kami** [comment]
Hm, what happened to #1575 - it didn't get merged into master and v0.11?

I've added some tests and other improvements there so need to make sure it gets merged. In any case, I will wait with merging and cherry-picking the commit because someone had a good reason to not merge it.

**Comment by Kami** [comment]
Edit: I checked git log, it looks like a different branch ended up being merge into 0.11.0. I will cherry pick test and other fixes on a new branch and get it merged into master.

**Comment by Kami** [comment]
I went ahead and synced #1575 with latest master so it can be merged into master now.

**Pull request merged**